### PR TITLE
Test on Django 4.2

### PR DESCRIPTION
**Pull request by adamchainz**

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -20,7 +20,14 @@ usage() {
 case "$CMD" in
     "test" )
         echo "Django version: $(python -m django --version)"
-        python -m django test django_ratelimit "$@"
+        python \
+            -W error::ResourceWarning \
+            -W error::DeprecationWarning \
+            -W error::PendingDeprecationWarning \
```
Comment (benjaoming) [inline run.sh:26]: Is this only targeted for the Django 4.2 tests? Just thinking that it's a shame in Django 4.3 to miss new warnings.

**Review by benjaoming — APPROVED** [review]

**Comment by adamchainz** [comment]
There are two expected test runs which are the ones that I've removed, so they need removing from the branch protection rule.

File: run.sh (hunk 1 above)
Comment (adamchainz) [reply run.sh:26]: Django's `RemovedInDjangoXYWarning` warnings are floating between `DeprecationWarning` and `PendingDeprecationWarning` depending on the current version: https://github.com/django/django/blob/main/django/utils/deprecation.py#L7-L12

(Also there won't be a Django 4.3, just a 5.0)

Comment (benjaoming) [reply run.sh:26]: Ah! I read it wrongly, I'm too used to these patterns always being about silencing warnings. This is the thing we want :100:

**Review by jsocol — APPROVED** [review]

**Comment by jsocol** [comment]
Thank you for this! Also adjusted the branch protection rules to remove Dj3.2/Py3.11 and Dj4.0/Py3.11

**Pull request merged**